### DISCUSSION TITLE
can  ESP_WiFiManager_Lite be used for managing configuration in non-wifi setups

### DISCUSSION THREAD

**Discussion by rmallah** [khoih-prog/ESP_WiFiManager_Lite#9] [Q&A] [answered]
We want to use Wifi primarily and GSM (as fallback connection).
We want to store server hostname and token as config variable
and also a flag to indicate if we want to use Wifi or GSM for connectivity.

It appears that unless a valid wifi credentials is provided ESP_WiFiManager_Lite 
does not store the config variables ( is it correct?)

**Comment by rmallah** [comment]
I think i am nearing the solution.
A part of the solution is to set RESET_IF_NO_WIFI  to false in
defines.h
```
#define RESET_IF_NO_WIFI              false
```

**Comment by rmallah** [comment]
use case elaboration:

- We have 2 means of connectivity in the device namely GSM (via Sim800L) and ESP8266 WiFi
- The preference to be stored in a dynamic variable when can be configured from config portal.
- If the preference is to use GSM , Wifimulticonnect should not be invoked and interrupt the main function of the device.
- If the preference is to use Wifi , Wifimulticonnect should keep trying to connect using AP and Pass specified in config.
- If the enduser wants to reconfigure the device to change preference he can use multi-reset.

My Question is how can i achieve above and which is a better starting point ESP_WiFiManager_Lite  or ESP_WiFiManager?

**Comment by khoih-prog** [comment, marked as answer]
I'm sorry nobody can help you to do your job, especially with your special use case.

You can have a look at my [**BlynkGSM_Manager**](https://github.com/khoih-prog/BlynkGSM_Manager) and [**Blynk_Async_GSM_Manager**](https://github.com/khoih-prog/Blynk_Async_GSM_Manager) libraries which provide very similar requested features

> This is also a Blynk and WiFiManager Library for configuring/auto(re)connecting ESP8266/ESP32 modules to the available MultiWiFi APs and MultiBlynk servers at runtime. Connection is with or without SSL. Configuration data to be saved in either LittleFS / SPIFFS or EEPROM. Default Credentials as well as Dynamic custom parameters can be added and modified easily without coding knowledge. DoubleResetDetector is used to force Config Portal opening even if the Credentials are still valid.
> 
> - This is the new library, designed to help you eliminate hardcoding your Blynk credentials in ESP32 and ESP8266 boards using GSM shield (SIM800, SIM900, etc).
> 
> - You can update GSM Modem and Blynk Credentials any time you need to change via Configure Portal. Data are saved in LittleFS / SPIFFS or configurable locations in EEPROM.

That's all and you have to do some **research, think and write the code** to adapt the libraries **yourself**. I won't spend anymore time on this.

Good Luck,

**Reply by rmallah** [reply]
I am very pleased to tell that that by banking on excellent libraries by you like MRD,DRD ,and this library 
i have been able to wire towards a solution. Thanks for the suggestion on Blynk but we are coding something 
similar on server side.
